Tasks/GooglePlayReleaseV4/GooglePlay.ts

```diff
@@ -0,0 +1,489 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as glob from 'glob';
+
+import * as googleutil from './googleutil';
+import * as metadataHelper from './metadataHelper';
+
+import * as googleapis from 'googleapis';
+import { androidpublisher_v3 as pub3 } from 'googleapis';
+
+type Action = 'OnlyStoreListing' | 'SingleBundle' | 'SingleApk' | 'MultiApkAab';
+
+async function run(): Promise<void> {
+    try {
+        tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+        tl.debug('Prepare task inputs.');
+
+        // Authentication inputs
+
+        const authType: string = tl.getInput('authType', true);
+        let key: googleutil.ClientKey = {};
+        if (authType === 'JsonFile') {
+            const serviceAccountKeyFile: string = tl.getPathInput('serviceAccountKey', true, true);
+
+            const stats: tl.FsStats = tl.stats(serviceAccountKeyFile);
+            if (stats && stats.isFile()) {
+                key = require(serviceAccountKeyFile);
+            } else {
+                tl.debug(`The service account file path ${serviceAccountKeyFile} points to a directory.`);
+                throw new Error(tl.loc('InvalidAuthFile', serviceAccountKeyFile));
+            }
+        } else if (authType === 'ServiceEndpoint') {
+            let serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), false);
+            key.client_email = serviceEndpoint.parameters['username'];
+            key.private_key = serviceEndpoint.parameters['password'].replace(/\\n/g, '\n');
+        }
+
+        // General inputs
+
+        const actionString: string = tl.getInput('action', false);
+        if (
+            actionString !== 'MultiApkAab'
+            && actionString !== 'SingleBundle'
+            && actionString !== 'SingleApk'
+            && actionString !== 'OnlyStoreListing'
```

Comment:
I believe it's better to create a set of const with all types for validating action types instead of hardcoding these checks. Also, could it make to export this validation to a separate function?

Reply:
Fixed in https://github.com/microsoft/google-play-vsts-extension/pull/292/commits/8c6474660f36a9188d904a26f99fe6053d20d5bc